test/api_spec.js

```diff
@@ -115,91 +137,83 @@ describe("API Tests", function () {
             expect(error).toBe(null);
             expect(res.statusCode).toEqual(201);
             expect(res.body).toEqual('');
-            var route = proxy.routes['/'];
-            expect(route.target).toEqual(target);
-            expect(typeof route.last_activity).toEqual('object');
-            done();
+
+            proxy._routes.get("/", function (route) {
+                expect(route.target).toEqual(target);
+                expect(typeof route.last_activity).toEqual('object');
+                done();
+            });
         });
     });
 
     it("DELETE /api/routes[/path] deletes a route", function (done) {
         var port = 8998;
         var target = 'http://127.0.0.1:' + port;
         var path = '/user/bar';
-        util.add_target(proxy, path, port);
-        expect(proxy.routes[path].target).toEqual(target);
-        r.del(api_url + path, function (error, res, body) {
+
+        util.add_target(proxy, path, port, null, null, function () {
+            proxy._routes.get(path, function (route) {
+                expect(route.target).toEqual(target);
+
+                r.del(api_url + path, function (error, res, body) {
+                    expect(error).toBe(null);
+                    expect(res.statusCode).toEqual(204);
+                    expect(res.body).toEqual('');
+
+                    proxy._routes.get(path, function (deleted_route) {
+                        expect(deleted_route).toBe(undefined);
+                        done();
+                    });
+                });
+            });
+        });
+    });
+
+    it("GET /api/routes?inactive_since= with bad value returns a 400", function (done) {
+        r.get(api_url + "?inactive_since=endoftheuniverse", function (error, res, body) {
             expect(error).toBe(null);
-            expect(res.statusCode).toEqual(204);
-            expect(res.body).toEqual('');
-            expect(proxy.routes[path]).toBe(undefined);
+            expect(res.statusCode).toEqual(400);
             done();
         });
     });
 
     it("GET /api/routes?inactive_since= filters inactive entries", function (done) {
         var port = 8998;
         var path = '/yesterday';
-        util.add_target(proxy, '/yesterday', port);
-        util.add_target(proxy, '/today', port+1);
 
-        var now = new Date();
-        var yesterday = new Date(now.getTime() - (24 * 3.6e6));
-        var long_ago = new Date(1);
-        var hour_ago = new Date(now.getTime() - 3.6e6);
+        var now           = new Date();
+        var yesterday     = new Date(now.getTime() - (24 * 3.6e6));
+        var long_ago      = new Date(1);
+        var hour_ago      = new Date(now.getTime() - 3.6e6);
```

Comment:
I haven't done aligned assignments generally, and would prefer them not to be applied to existing code without any other changes. Not a big deal, just something for the future.